tests/scripts/audit-validity-dates.py

```diff
@@ -65,8 +65,13 @@ class AuditData:
     #pylint: disable=too-few-public-methods
     def __init__(self, data_type: DataType, x509_obj):
         self.data_type = data_type
-        self.location = ""
+        # the locations that the x509 object could be found
+        self.locations = [] # type: typing.List[str]
         self.fill_validity_duration(x509_obj)
+        self._obj = x509_obj
+
+    def __eq__(self, __value) -> bool:
```

Comment:
Overriding equality can have non-obvious effects. Can you please add a comment that explains at least why we're doing this?

If I understand correctly, at least in the commit "cert_audit: Merge audit_data for identical X.509 objects", overriding equality allows `merge_auditdata` to use `in` to locate objects in a list. This could easily be done by checking `any(r._obj == x._obj for x in results)` instead. Though using sets might be a justification for overriding `__eq__`, but also `__hash__`.

Reply:
> If I understand correctly, at least in the commit "cert_audit: Merge audit_data for identical X.509 objects", overriding equality allows merge_auditdata to use in to locate objects in a list. 

Yes, this is exactly what I want to do. I will re-consider this part of code.

> https://github.com/Mbed-TLS/mbedtls/pull/7512#discussion_r1180521803:
Since we're merging identical objects, should the data structure be a set rather than a list?
Or should it be a dictionary mapping X.509 objects to a list/set of locations?

> Though using sets might be a justification for overriding `__eq__`, but also `__hash__`.

Yes, I have tried with set, the problem was the underlying `crl` object in `cryptography` is not hashable. Dictionary should be a good option, but I have no idea what should be used as key, maybe the X.509 objects? What's your suggestion?

Reply:
> but I have no idea what should be used as key, maybe the X.509 objects? 

Since crl object is not hashable, it couldn't be used as dictionary key either.

Reply:
> the underlying crl object in cryptography is not hashable

Ah, if the underlying object is comparable for equality but not hashable, then indeed we can't use a dict. Unless we hash on the DER representation?

I'd be happy with either solution:

* Define a custom hash based on the DER (or something equivalent).
* A comment that explains that the objects aren't hashable, so we can't use data structures like dictionaries on them.